### PR TITLE
LF-3004: Update Unit component

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- create a file for `useReactSelectStyles` hook
- add more logic to `useUnit` hook

(I am working on refactoring the Unit component, and I plan to do it in 3+ PRs. The PRs I am creating are going to be merged to the "LF-2995+3002/Unit_displays" branch, and when everything is ready, the branch is going to be merged to the integration branch.)
- [x] LF-3003 Extract logic from Unit component and create a hook
- [x] **LF-3004 Update Unit component to make it work as a presentation component**
- [ ] LF-3005 Fix useUnit hook

Jira link: https://lite-farm.atlassian.net/browse/LF-3004

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by JimmyRowland — APPROVED** [review]

File: packages/webapp/src/components/Form/Unit/useUnit.jsx [hunk 1]
```diff
@@ -59,32 +73,52 @@ const useUnit = ({
...
+          ...getDefaultUnit(unitType, value, system, databaseUnit),
+          options,
+          databaseUnit,
+          defaultIsSelectDisabled,
+          measure,
+          reactSelectWidth,
+          dividerWidth,
+          onKeyDown,
+        };
+  }, []);
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Form/Unit/useUnit.jsx:121]: @SayakaOno 
useMemo should have dependency list right?

If dependency list is big should useMemo be broken up into several useMemo so it is not re-run often?

Comment (SayakaOno) [reply packages/webapp/src/components/Form/Unit/useUnit.jsx:121]: This useMemo returns default values for the Unit component, so it should only be called once. I will add some comments in the next PR to make it clear!

Comment (Duncan-Brain) [reply packages/webapp/src/components/Form/Unit/useUnit.jsx:121]: I will look into it more for next PR. But only calling it once seems to defeat the purpose of useMemo.

**Review by Duncan-Brain — APPROVED** [review]
Looks like a presentation component to me

File: packages/webapp/src/components/Form/Unit/useUnit.jsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Form/Unit/useUnit.jsx:121]: I understand what you mean. Please let me know if you come up with a better solution!
Thank you for reviewing :)

**Pull request merged**